miden/src/cli/prove.rs

```diff
@@ -139,20 +152,53 @@ impl ProveCmd {
     }
 }
 
-// HELPER FUNCTIONS
-// ================================================================================================
-
 #[instrument(skip_all)]
 fn load_data(params: &ProveCmd) -> Result<(Program, InputFile), Report> {
```

Comment:
Similar comment as above - I think we can split this up into two separate functions.